### PR TITLE
[#3701]refactor(API): Refactoring SupportsSchemas.listSchemas() method to return String[]

### PR DESCRIPTION
**Pull request by shaofengshi**
### What changes were proposed in this pull request?

Currently, the SupportsSchemas.listSchemas() method returns an array of NameIdentifier to represents the schemas. Actually a String[] will be more clear and easier to use.

### Why are the changes needed?

To make the API more clear to use.

Fix: #3701

### Does this PR introduce _any_ user-facing change?

Almost not; Just return type changed, will be more simple.

### How was this patch tested?

Yes, many test cases cover this API.

### CODE REVIEW

**Comment by shaofengshi** [comment]
@yuqi1129 @mchades please review this small API change; The update for doc will be made a little bit later in batch after the code be merged.

**Comment by mchades** [comment]
Missing changes in Python client?

File: api/src/main/java/com/datastrato/gravitino/SupportsSchemas.java [hunk 1]
```diff
@@ -39,12 +39,12 @@ public interface SupportsSchemas {
    *
    * <p>If an entity such as a table, view exists, its parent schemas must also exist and must be
    * returned by this discovery method. For example, if table a.b.t exists, this method invoked as
-   * listSchemas(a) must return [a.b] in the result array
+   * listSchemas(a) must return [b] in the result array
    *
-   * @return An array of schema identifier under the namespace.
+   * @return An array of schema names under the namespace.
```
Comment (mchades) [inline api/src/main/java/com/datastrato/gravitino/SupportsSchemas.java:44]: The comments in the implementation class also need to be modified：
https://github.com/datastrato/gravitino/blob/c8dabb745e7cb7b27573089b85af96c12a71e45a/clients/client-java/src/main/java/com/datastrato/gravitino/client/BaseSchemaCatalog.java#L70

Comment (shaofengshi) [reply api/src/main/java/com/datastrato/gravitino/SupportsSchemas.java:44]: Thx! I just updated the java doc in BaseSchemaCatalog.java, please check;

**Comment by shaofengshi** [comment]
> Missing changes in Python client?

The change in Python will be tracked with another issue: https://github.com/datastrato/gravitino/issues/3732

**Comment by mchades** [comment]
Overall LGTM

File: catalogs/catalog-jdbc-doris/src/test/java/com/datastrato/gravitino/catalog/doris/integration/test/CatalogDorisIT.java [hunk 2]
```diff
@@ -185,20 +184,16 @@ void testDorisSchemaBasicOperation() {
     SupportsSchemas schemas = catalog.asSchemas();
 
     // test list schemas
-    NameIdentifier[] nameIdentifiers = schemas.listSchemas();
-    Set<String> schemaNames =
-        Arrays.stream(nameIdentifiers).map(NameIdentifier::name).collect(Collectors.toSet());
-    Assertions.assertTrue(schemaNames.contains(schemaName));
+    String[] schemaNames = schemas.listSchemas();
+    Assertions.assertTrue(Arrays.asList(schemaNames).contains(schemaName));
```
Comment (yuqi1129) [inline catalogs/catalog-jdbc-doris/src/test/java/com/datastrato/gravitino/catalog/doris/integration/test/CatalogDorisIT.java:188]: Maybe you can check if you can use `ArrayUtils.contains()` without introducing extra dependencies.

File: clients/client-java/src/main/java/com/datastrato/gravitino/client/BaseSchemaCatalog.java [hunk 3]
```diff
@@ -81,7 +81,7 @@ public NameIdentifier[] listSchemas() throws NoSuchCatalogException {
             ErrorHandlers.schemaErrorHandler());
     resp.validate();
 
-    return resp.identifiers();
+    return Arrays.stream(resp.identifiers()).map(NameIdentifier::name).toArray(String[]::new);
```
Comment (yuqi1129) [inline clients/client-java/src/main/java/com/datastrato/gravitino/client/BaseSchemaCatalog.java:84]: So, will the server side of this interface remain the same?

**Comment by jerryshao** [comment]
LGTM.

File: catalogs/catalog-jdbc-doris/src/test/java/com/datastrato/gravitino/catalog/doris/integration/test/CatalogDorisIT.java (hunk 2 above)
Comment (shaofengshi) [reply catalogs/catalog-jdbc-doris/src/test/java/com/datastrato/gravitino/catalog/doris/integration/test/CatalogDorisIT.java:188]: I don't understand; Arrays is the JDK utility class, while ArrayUtils is apache commons lang; As this is so simple a case, I don't think it is necessary to use ArrayUtils here.

Comment (yuqi1129) [reply catalogs/catalog-jdbc-doris/src/test/java/com/datastrato/gravitino/catalog/doris/integration/test/CatalogDorisIT.java:188]: This is a personal preference and the current implementation looks good.

File: clients/client-java/src/main/java/com/datastrato/gravitino/client/BaseSchemaCatalog.java (hunk 3 above)
Comment (yuqi1129) [reply clients/client-java/src/main/java/com/datastrato/gravitino/client/BaseSchemaCatalog.java:84]: Can you respond to it?

Comment (shaofengshi) [reply clients/client-java/src/main/java/com/datastrato/gravitino/client/BaseSchemaCatalog.java:84]: Right, server side won't change; The "SupportsSchemas" has two version, one is for client, the other is for server side. This change has already been made in the main branch.

File: catalogs/catalog-jdbc-doris/src/test/java/com/datastrato/gravitino/catalog/doris/integration/test/CatalogDorisIT.java (hunk 2 above)
Comment (shaofengshi) [reply catalogs/catalog-jdbc-doris/src/test/java/com/datastrato/gravitino/catalog/doris/integration/test/CatalogDorisIT.java:188]: ok, thanks~

**Review by yuqi1129 — APPROVED** [review]
LGTM

**Pull request merged**